closed/src/jdk.crypto.ec/share/classes/sun/security/ec/NativeECDHKeyAgreement.java

```diff
@@ -129,6 +130,8 @@ protected void engineInit(Key key, SecureRandom random)
         } else {
             this.javaImplementation = null;
         }
+
+        ECUtil.checkPrivateKey(this.privateKey);
```

Comment:
Is there a reason this couldn't (or shouldn't) happen earlier, before line 107?

Reply:
moved the check to line 107